crates/mentat-syncer/src/syncer.rs

```diff
@@ -0,0 +1,607 @@
+//! handles the synchronizing of blocks from a node
+
+use std::{
+    mem::size_of_val,
+    sync::{
+        atomic::{AtomicUsize, Ordering},
+        Arc,
+        Weak,
+    },
+    thread::{sleep, spawn},
+};
+
+use crossbeam_channel::{bounded, Receiver, Sender};
+use indexmap::IndexMap;
+use mentat_types::{hash, Block, BlockIdentifier, EstimateSize, PartialBlockIdentifier};
```

Comment:
NIT: the commonly used imports like mentat_types, some std::sync stuff, etc can all be imported from super to these repetitive imports across files.